var/spack/repos/builtin/packages/libdwarf/package.py

```diff
@@ -77,23 +79,44 @@ def install(self, spec, prefix):
                 if spec.satisfies('^libelf'):
                     libelf_inc_dir = join_path(spec['libelf'].prefix,
                                                'include/libelf')
-                    extra_config_args.append('CFLAGS=-I{0}'.format(
-                                             libelf_inc_dir))
+                    extra_config_args.append(
+                        'CFLAGS=-I{0} -Wl,-L{1} -Wl,-lelf'.format(
+                            libelf_inc_dir, spec['libelf'].prefix.lib))
                 configure("--prefix=" + prefix, "--enable-shared",
                           *extra_config_args)
+                filter_file(r'^dwfzlib\s*=\s*-lz',
+                            'dwfzlib=-L{0} -lz'.format(
+                                self.spec['zlib'].prefix.lib),
+                            'Makefile')
                 make()
 
+                libdwarf_name = 'libdwarf.{0}'.format(dso_suffix)
                 install('libdwarf.a',  prefix.lib)
-                install('libdwarf.so', prefix.lib)
+                install('libdwarf.so', join_path(prefix.lib, libdwarf_name))
                 install('libdwarf.h',  prefix.include)
                 install('dwarf.h',     prefix.include)
 
+                # It seems like fix_darwin_install_name can't be used
+                # here directly; the install name of the library in
+                # the stage directory must be fixed in order for dyld
+                # to locate it on Darwin when spack builds dwarfdump
+                if sys.platform == 'darwin':
+                    install_name_tool = which('install_name_tool')
```

Comment:
same as in another PR, please use `fix_darwin_install_name`

Reply:
hm, looks like this is a peculiar case where libs have `.so` suffix even on darwin...

Reply:
how about adding a second parameter to `fix_darwin_install_name` with default `suffix=dylib` so that you can use it here as well?

Reply:
nevermind, I completely missed your comment above

Reply:
> same as in another PR, please use `fix_darwin_install_name`

That method won't work here because: (1) the build system builds the shared library with the wrong suffix, so it won't change the file, (2) the correct path for this library isn't the full path, but a relative path, due to a quirk in the build system. The comment immediately above the `if` clauses attempts to note these two issues.